File: langgraph/result/qwen/7b-eq2js/57.js
function checkMonotonicSequence(sequence) {
    let ascending = [...sequence];
    let descending = [...sequence];
    
    ascending.sort((a, b) => a - b);
    descending.sort((a, b) => b - a);
    
    if (JSON.stringify(sequence) === JSON.stringify(ascending) || JSON.stringify(sequence) === JSON.stringify(descending)) {
        return true;
    }
    return false;
}

File: langgraph/data/semantic_equiv/57.py
def check_monotonic_sequence(sequence: list):
    ascending = all(x <= y for x, y in zip(sequence, sequence[1:]))
    descending = all(x >= y for x, y in zip(sequence, sequence[1:]))

    return ascending or descending
# method name: monotonic -> check_monotonic_sequence
# parameter name: l -> sequence
# replaced sorting with direct comparison using zip and all()
# used two separate variables for ascending and descending checks